kclvm/tools/src/LSP/src/goto_def.rs

```diff
@@ -126,8 +127,38 @@ fn goto_def_for_import(
     program: &Program,
 ) -> Option<GotoDefinitionResponse> {
     let pkgpath = &stmt.path;
-    let real_path =
+    let mut real_path =
         Path::new(&program.root).join(pkgpath.replace('.', &std::path::MAIN_SEPARATOR.to_string()));
+    let mut positions = get_pos_from_real_path(&real_path);
+
+    if positions.len() == 0 && !real_path.exists() {
+        real_path = PathBuf::new();
+
+        let pkg_root = fetch_metadata(program.root.clone().into())
+            .map(|metadata| {
+                metadata
+                    .packages
+                    .get(&stmt.pkg_name)
+                    .map_or(PathBuf::new(), |pkg| pkg.manifest_path.clone())
+            })
+            .unwrap_or_else(|_| PathBuf::new());
+
+        real_path = real_path.join(pkg_root);
+
+        let pkgpath = match rm_external_pkg_name(pkgpath) {
+            Ok(path) => path,
+            Err(_) => String::new(),
+        };
+        pkgpath.split('.').for_each(|s| real_path.push(s));
+    }
+    positions = get_pos_from_real_path(&real_path);
+
+    positions_to_goto_def_resp(&positions)
+}
+
+fn get_pos_from_real_path(
```

Comment:
Move this function into the util crates.

Reply:
fixed.